result.go

```diff
@@ -225,6 +225,77 @@ func (q Query) FirstWithT(predicateFn interface{}) interface{} {
 	return q.FirstWith(predicateFunc)
 }
 
+// ForEach performs the specified action on each element of a collection.
+func (q Query) ForEach(action func(interface{})) {
```

Comment:
@kalaninja Awesome! What if we change the action signature to return an error? 
```go
func (q Query) ForEach(action func(interface{}) error) err error {
  next := q.Iterate()
   
  for item, ok := next(); ok; item, ok = next() {
    if err = action(item); err != nil{
      break
    }
  }
  return
}
```
if the action returns an error, ForEach breaks the loop and return the error.